### ISSUE TITLE
Benchmark mode with command line switches

### ISSUE THREAD

**Issue by cb88** [OpenRCT2/OpenRCT2#6339]
Would it be possible to have a  benchmark mode and possibly an included torture test save file?

I see there is already benchgfx as a command line option but it doesn't really render output it seems to just load the map and perhaps render it off screen for one frame?

A lot of computer hardware sites use features of games like this to demo new hardware and OpenRCT2 seems to be capable of being a fairly heavy graphical and CPU heavy load given a sufficiently complex map.

**Comment by janisozaur** [comment]
I don't think this is worth implementing. We have benchgfx for some rudimentary checks that are somewhat reproducible between two builds, but I don't see OpenRCT2 as a proper benchmark tool.

The issues are:
1. It requires external assets,
2. For it to be useful as a benchmark tool, it would have to check some known component. We a) don't know what it tests _exactly_ b) don't quite care,
3. There are better tools that highlight specific parts of SUT,
4. The ever-changing nature of OpenRCT2 does not make it a suitable candidate as a benchmark utility.

There's still place (and need) for some other benchmarking that we could use for our own purposes.

I'm tempted to close this as `wontfix`, but would like to hear from other team members before I do.

**Comment by Gymnasiast** [comment]
I was going to say, don't we have a rudimentary benchmark already? We also have `bpb.sv6` in the tests folder, which is quite a full park and could serve the purpose of a torture test (though any NE park is probably better in that respect).

**Comment by IntelOrca** [comment]
> ...an included torture test save file?

There is a particular park we frequently use to stress test the game. It is a park where everything is inside a big dome made out of glass scenery. We could include it in the code base potentially, I am not sure how many custom objects it uses though - that could be an issue.

> I see there is already benchgfx as a command line option but it doesn't really render output it seems to just load the map and perhaps render it off screen for one frame?

It renders the game, it just doesn't store it. Why does it need to?

**Comment by cb88** [comment]
"4. The ever-changing nature of OpenRCT2 does not make it a suitable candidate as a benchmark utility."
That is the same case with basically any game, so not really a valid point.  I think external assets are also pretty much a non issue as anyone that cares will have them, they are on the other hand pretty static since it is an old game. 

"It renders the game, it just doesn't store it. Why does it need to?"
Wouldn't it? The fact that it isn't drawing to the screen... probably means there are components of the game that aren't exercised. I guess it kinda makes sense for the software rasterizer... but it appears that it can benchmark the other modes as well (it changed it's output based on which renderer I had configured in the game).

**Comment by ZehMatt** [comment]
To provide proper benchmarks you would need to setup a deterministic scenario. What I would do is something such as, load the park, set the camera to {0,0,0}, set zoom to 0, rotation to 0, and profile that for lets say 200 game ticks, then change the camera to rotation 1, profile 200 ticks again and basically repeat this step with all rotations, zooms.
